send/send.c

```diff
@@ -1475,6 +1475,7 @@ struct Address *mutt_default_from(struct ConfigSubset *sub)
   else if (c_use_domain)
   {
     struct Address *addr = mutt_addr_new();
+    addr->mailbox = buf_new(NULL);
```

Comment:
Perhaps we should first come up with a string for the mailbox, wither this or `Username` as in the else branch, then call `mutt_addr_create` outside the branch?

Reply:
how about (untested)

```c
struct Address *mutt_default_from(struct ConfigSubset *sub)
{
  const struct Address *c_from = cs_subset_address(sub, "from");
  if (c_from)
  {
    return mutt_addr_copy(c_from);
  }

  char domain[1024] = { 0 };
  const char *mailbox = Username;
  const bool c_use_domain = cs_subset_bool(sub, "use_domain");
  if (c_use_domain)
  {
    snprintf(domain, sizeof(domain), "%s@%s", NONULL(Username), NONULL(mutt_fqdn(true, sub)));
    mailbox = domain;
  }

  return mutt_addr_create(NULL, domain);
}
```